File: py-client/core/model/queue.py
import uuid
from enum import Enum


class Queue:
    class Type(Enum):
        SIMPLE = 1
        MULTIPLE = 2

    def __init__(self, code: str, name: str, guid: str = uuid.uuid4(), pending_messages_count: int = 0):
        self.guid = guid
        self.code = code
        self.name = name
        self.pending_messages_count = pending_messages_count

    def __str__(self):
        return f"""
            Queue {self.code} 
                id={self.guid}
                name={self.name}
                Has {self.pending_messages_count} pending messages
        """

File: client/core/domain/src/main/kotlin/io/github/vinicreis/pubsub/client/core/model/Queue.kt
package io.github.vinicreis.pubsub.client.core.model

import java.util.*

data class Queue(
    val id: UUID,
    val code: String,
    val name: String = code,
    val type: Type,
    val pendingMessagesCount: Long,
) {
    enum class Type {
        SIMPLE,
        MULTIPLE;

        companion object {
            fun fromOrdinal(ordinal: Int) = Type.entries.firstOrNull { it.ordinal == ordinal }
        }
    }
}
